lib/resources/gen/goals.js

```diff
@@ -0,0 +1,282 @@
+ * @param {Object} [dispatchOptions]: Options, if any, to pass the dispatcher for the request
+ * @return {Promise} The requested resource
+ */
+Goals.prototype.getSubgoalsForGoal = function(
+    params,
+    dispatchOptions
+) {
+    var path = "/goals/{goal_gid}/subgoals";
+
+    return this.dispatchGetCollection(path, params, dispatchOptions)
+};
+
+
+/**
+ * Remove a collaborator from a goal
+ * @param {Object} data: Data for the request
+ * @param {Object} [dispatchOptions]: Options, if any, to pass the dispatcher for the request
+ * @return {Promise} The requested resource
+ */
+Goals.prototype.removeFollowers = function(
+    data,
+    dispatchOptions
+) {
+    var path = "/goals/{goal_gid}/removeFollowers";
+
+    return this.dispatchPost(path, data, dispatchOptions)
+};
+
+
+/**
+ * Remove a subgoal from a goal
+ * @param {Object} data: Data for the request
+ * @param {Object} [dispatchOptions]: Options, if any, to pass the dispatcher for the request
+ * @return {Promise} The requested resource
+ */
+Goals.prototype.removeSubgoal = function(
+    data,
+    dispatchOptions
+) {
+    var path = "/goals/{goal_gid}/removeSubgoal";
+
+    return this.dispatchPost(path, data, dispatchOptions)
+};
+
+
+/**
+ * Remove a project/portfolio as supporting work for a goal.
+ * @param {Object} data: Data for the request
+ * @param {Object} [dispatchOptions]: Options, if any, to pass the dispatcher for the request
+ * @return {Promise} The requested resource
+ */
+Goals.prototype.removeSupportingWorkForGoal = function(
+    data,
+    dispatchOptions
+) {
+    var path = "/goals/{goal_gid}/removeSupportingWork";
+
+    return this.dispatchPost(path, data, dispatchOptions)
+};
+
+
+/**
+ * Get supporting work from a goal
+ * @param {Object} params: Parameters for the request
+     - optFields {[String]}:  Defines fields to return. Some requests return *compact* representations of objects in order to conserve resources and complete the request more efficiently. Other times requests return more information than you may need. This option allows you to list the exact set of fields that the API should be sure to return for the objects. The field names should be provided as paths, described below. The id of included objects will always be returned, regardless of the field options.
+     - optPretty {Boolean}:  Provides “pretty” output. Provides the response in a “pretty” format. In the case of JSON this means doing proper line breaking and indentation to make it readable. This will take extra time and increase the response size so it is advisable only to use this during debugging.
+ * @param {Object} [dispatchOptions]: Options, if any, to pass the dispatcher for the request
+ * @return {Promise} The requested resource
+ */
+Goals.prototype.supportingWork = function(
```

Comment:
🧐 Nit: I think this is fine, but my first guess as to what this function would be called is getSupportingWorkForGoal (which matches the signature of addSupportingWorkForGoal and removeSupportingWorkForGoal). I have to admit, though, that I could be out of date on that - Ross and I were discussing naming conventions for our client libraries ages ago and I might misremember the convention we settled on.

Reply:
Make sense! I think we just wanted to be consistent with how the naming convention went for past resources. Generally, `get` requests are just the names of the resources themselves.